web-services/query/src/main/java/datawave/webservice/query/metric/QueryMetricsWriter.java

```diff
@@ -128,7 +135,7 @@ public void onRefresh(@Observes RefreshEvent event, BeanManager bm) {
 
     public void addMetricToQueue(QueryMetricHolder queryMetricHolder) {
         try {
-            if (shuttingDown) {
+            if (shutDownQueue) {
                 log.error(String.format("shutting down - dropping metric for queryId %s", queryMetricHolder.getQueryMetric().getQueryId()));
             } else if (!blockingQueue.offer(queryMetricHolder)) {
                 if (queryMetricHolder.getQueryMetric().getLifecycle().compareTo(Lifecycle.INITIALIZED) <= 0) {
```

Comment:
Just curious why we do not log dropping a metric even when we are not yet initialized?

Reply:
1) During shutdown of the W/S, shutDownQueue gets set to true first to prevent any more metrics from being added.  At that point, log metric updates that are dropped.  If all queries were completed, there shouldn't be any.

2) The else if clause is for if we fail adding a metric update to the queue (queue is full).  This PR and changes to the metric service are meant to prevent this situation, but if it happens (again) then we would print a log message for every query event (create, initialized, every page, closed, etc).  The if statement to only log a metric update when the Lifecycle is <= INITIALIZED (i.e. CREATED, INITIALIZED) is to limit the flood of logs while still alerting us that the queue is full and we are dropping metrics.

I'm open to suggestions, but the hope is that in adding threads on the W/S side to hit http receiver threads on the metric service side that we avoid this happening.